src/Raven.Server/Documents/Queries/Graph/EdgeQueryStep.cs

```diff
@@ -65,9 +65,16 @@ public bool IsEmpty()
             return _results.Count == 0;
         }
 
+        public bool CollectIntermediateResults { get; set; }
+
+        public List<Match> IntermediateResults => CollectIntermediateResults?_results: new List<GraphQueryRunner.Match>();
```

Comment:
I don't like this. It just allocates and throw it away.
Why do we need this to be valid at all if `CollectIntermediateResults` is `false`?

Reply:
we could return null but I would still want an indication that the intermediate results are needed, see: https://issues.hibernatingrhinos.com/issue/RavenDB-12572